src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -74,9 +74,6 @@ private void RefreshAuthToken()
 
         private KustoConnectionStringBuilder GetKustoConnectionStringBuilder(DataSourceConnectionDetails connectionDetails)
         {
-            ValidationUtils.IsTrue<ArgumentException>(!string.IsNullOrWhiteSpace(connectionDetails.UserToken),
```

Comment:
Don't you still want this check if the auth type is `AzureMFA` or `dstsAuth`?

Reply:
Moved this into the mapping function.